src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -42,6 +42,14 @@ private static function prepareRequest(?string $method, ?string $url, array $opt
 
         $options = self::mergeDefaultOptions($options, $defaultOptions, $allowExtraOptions);
 
+        if (isset($options['buffer']) && \is_string($options['buffer'])) {
```

Comment:
```php
if (\is_string($options['buffer'] ?? null) && false === @preg_match($options['buffer'], '')) {
    throw new InvalidArgumentException('Option "buffer" should be a valid regular expression.');
}
```

Reply:
Replaced the check and changed the exception message